closed/src/java.base/share/classes/com/sun/crypto/provider/NativeChaCha20Cipher.java

```diff
@@ -558,25 +573,18 @@ private void init(int opmode, Key key, byte[] newNonce)
         }
 
         // Make sure that the provided key and nonce are unique before
-        // assigning them to the object.
+        // assigning them to the object.  Key and nonce uniqueness
+        // protection is for encryption operations only.
```

Comment:
I think it makes more sense for this comment to be inside the `if` statement beginning on line 579.
Also, I suggest you update the comment for `checkKeyAndNonce()` to mention that the check is only relevant for encryption mode.

Reply:
Moved it and added a mention in the method description.